src/mongo_logger.ts

```diff
@@ -122,58 +178,262 @@ function resolveLogPath(
+            break;
+          case 'error':
+            log.reason = 'An error occurred while using the connection';
+            if (ev.error) {
+              log.error = ev.error;
+            }
+            break;
+          case 'poolClosed':
+            log.reason = 'Connection pool was closed';
+            break;
+          default:
+          // Omit if we have some other reason as it would be invalid
+        }
+        break;
+      case CONNECTION_CHECK_OUT_STARTED:
+        ev = logObject as ConnectionCheckOutStartedEvent;
+        log.message = 'Connection checkout started';
+        break;
+      case CONNECTION_CHECK_OUT_FAILED:
+        ev = logObject as ConnectionCheckOutFailedEvent;
+        log.message = 'Connection checkout failed';
+        log.reason = ev.reason;
+        break;
+      case CONNECTION_CHECKED_OUT:
+        ev = logObject as ConnectionCheckedOutEvent;
+        log.message = 'Connection checked out';
+        log.driverConnectionId = ev.connectionId;
+        break;
+      case CONNECTION_CHECKED_IN:
+        ev = logObject as ConnectionCheckedInEvent;
+        log.message = 'Connection checked in';
+        log.driverConnectionId = ev.connectionId;
+        break;
+    }
+  } else {
+    for (const key in logObject) {
+      const value = logObject[key];
+      // eslint-disable-next-line no-restricted-syntax
+      if (value === undefined || value === null) continue;
+      log[key] = value;
+    }
+  }
+  return log;
 }
 
 /** @internal */
 export class MongoLogger {
   componentSeverities: Record<MongoLoggableComponent, SeverityLevel>;
   maxDocumentLength: number;
-  logDestination: Writable;
+  logDestination: MongoDBLogWritable | Writable;
+  writeStringifiedLogs: boolean;
 
   constructor(options: MongoLoggerOptions) {
     this.componentSeverities = options.componentSeverities;
     this.maxDocumentLength = options.maxDocumentLength;
     this.logDestination = options.logDestination;
+    this.writeStringifiedLogs =
+      this.logDestination === process.stderr || this.logDestination === process.stdout;
   }
 
-  /* eslint-disable @typescript-eslint/no-unused-vars */
-  /* eslint-disable @typescript-eslint/no-empty-function */
-  emergency(component: any, message: any): void {}
-
-  alert(component: any, message: any): void {}
-
-  critical(component: any, message: any): void {}
-
-  error(component: any, message: any): void {}
-
-  warn(component: any, message: any): void {}
-
-  notice(component: any, message: any): void {}
-
-  info(component: any, message: any): void {}
-
-  debug(component: any, message: any): void {}
+  /** @experimental */
+  emergency(component: MongoLoggableComponent, message: Loggable | string): void {
+    this.log(component, 'emergency', message);
+  }
 
-  trace(component: any, message: any): void {}
+  private log(
+    component: MongoLoggableComponent,
+    severity: SeverityLevel,
```

Comment:
Super minor, but I would've put severity first, component second, the thinking is that the sev is not _really_ a variable, all of log's invocations are going to have the first argument be a hardcoded string literal. 

Maybe it's worth keeping all our logger wrappers super slim? at least until there's a need for more than proxying the sev.
```ts
emergency = this.log.bind(this, 'emergency')
```